packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -927,10 +902,134 @@ class _EmptySliderComponentShape extends SliderComponentShape {
   }
 }
 
-// The following shapes are the material defaults.
+/// Base track shape with default sizing.
+///
+/// The [SliderTrackShape]s that use this base class are:
+///   [RectangularSliderTrackShape],
+///   [RoundedRectSliderTrackShape]
+///
+/// The height is set from [SliderThemeData.trackHeight] and the width is the
+/// width of the parent box, but with the largest shape subtracted out from
+/// [SliderThemeData.thumbShape] and [SliderThemeData.overlayShape].
+abstract class BaseSliderTrackShape {
+  /// Returns a rect that represents the track bounds that fits within the
+  /// [Slider].
+  ///
+  /// The width is the width of the [Slider], but padded by the max
+  /// of the overlay and thumb radius. The height is defined by the
+  /// [SliderThemeData.trackHeight].
+  ///
+  /// The rect is centered both horizontally and vertically within the slider
+  /// bounds.
+  Rect getPreferredRect({
+    RenderBox parentBox,
+    Offset offset = Offset.zero,
+    SliderThemeData sliderTheme,
+    bool isEnabled,
+    bool isDiscrete,
+  }) {
+    isEnabled = isEnabled ?? false;
```

Comment:
Can these 2 be set to false as default values in the param list instead of here?